src/AggregateFunctions/SingleValueData.cpp

```diff
@@ -0,0 +1,861 @@
+{
+    if (row_begin >= row_end)
+        return std::nullopt;
+
+    const auto & vec = assert_cast<const ColVecType &>(column);
+    if constexpr (has_find_extreme_implementation<T>)
+    {
+        std::optional<T> opt = findExtremeMax(vec.getData().data(), row_begin, row_end);
+        if (!opt || (has() && value >= opt))
+            return std::nullopt;
+
+        /// TODO: Implement findExtremeMaxIndex to do the lookup properly (with SIMD and batching)
+        for (size_t i = row_begin; i < row_end; i++)
+            if (vec.getData()[i] == *opt)
+                return i;
+        return row_end;
+    }
+    else
+    {
+        size_t index = row_begin;
+        for (size_t i = index + 1; i < row_end; i++)
+            if (vec.getData()[i] > vec.getData()[index])
+                index = i;
+        return index;
+    }
+}
+
+
+#if USE_EMBEDDED_COMPILER
+
+template <typename T>
+bool SingleValueDataFixed<T>::isCompilable(const IDataType & type)
+{
+    return canBeNativeType(type);
+}
+
+template <typename T>
+llvm::Value * SingleValueDataFixed<T>::getValuePtrFromAggregateDataPtr(llvm::IRBuilderBase & builder, llvm::Value * aggregate_data_ptr)
+{
+    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
+    auto * value_ptr = b.CreateConstInBoundsGEP1_64(b.getInt8Ty(), aggregate_data_ptr, value_offset);
+    return value_ptr;
+}
+
+template <typename T>
+llvm::Value * SingleValueDataFixed<T>::getValueFromAggregateDataPtr(llvm::IRBuilderBase & builder, llvm::Value * aggregate_data_ptr)
+{
+    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
+    auto * type = toNativeType<T>(builder);
+    auto * value_ptr = getValuePtrFromAggregateDataPtr(builder, aggregate_data_ptr);
+    return b.CreateLoad(type, value_ptr);
+}
+
+template <typename T>
+llvm::Value * SingleValueDataFixed<T>::getHasValuePtrFromAggregateDataPtr(llvm::IRBuilderBase & builder, llvm::Value * aggregate_data_ptr)
+{
+    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
+    auto * has_value_ptr = b.CreateConstInBoundsGEP1_64(b.getInt8Ty(), aggregate_data_ptr, has_value_offset);
+    return has_value_ptr;
+}
+
+template <typename T>
+llvm::Value * SingleValueDataFixed<T>::getHasValueFromAggregateDataPtr(llvm::IRBuilderBase & builder, llvm::Value * aggregate_data_ptr)
+{
+    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
+    auto * has_value_ptr = getHasValuePtrFromAggregateDataPtr(builder, aggregate_data_ptr);
+    return b.CreateLoad(b.getInt1Ty(), has_value_ptr);
+}
+
+template <typename T>
+void SingleValueDataFixed<T>::compileCreate(
+    llvm::IRBuilderBase & builder, llvm::Value * aggregate_data_ptr, size_t size_of_data, size_t alignof_data)
+{
+    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
+    b.CreateMemSet(aggregate_data_ptr, llvm::ConstantInt::get(b.getInt8Ty(), 0), size_of_data, llvm::assumeAligned(alignof_data));
```

Comment:
This is now broken since it's not setting up the virtual table (`SingleValueDataFixed` inherits from `SingleValueDataBase` so we can use a base class pointer) so we can't call any virtual method after. 


Looking at the bytecode for the a constructor emitted by the compiler it's something like this:
```
define weak_odr dso_local void @_ZN2DB20SingleValueDataFixedItEC2Ev(ptr noundef nonnull align 8 dereferenceable(16) %0) unnamed_addr #1 comdat($_ZN2DB20SingleValueDataFixedItEC5Ev) align 2 {
  %2 = alloca ptr, align 8
  store ptr %0, ptr %2, align 8
  %3 = load ptr, ptr %2, align 8
  call void @_ZN2DB19SingleValueDataBaseC2Ev(ptr noundef nonnull align 8 dereferenceable(8) %3) #15
  store ptr getelementptr inbounds ({ [22 x ptr] }, ptr @_ZTVN2DB20SingleValueDataFixedItEE, i32 0, inrange i32 0, i32 2), ptr %3, align 8
  %4 = getelementptr inbounds %"struct.DB::SingleValueDataFixed.127", ptr %3, i32 0, i32 1
  store i16 0, ptr %4, align 8
  %5 = getelementptr inbounds %"struct.DB::SingleValueDataFixed.127", ptr %3, i32 0, i32 2
  store i8 0, ptr %5, align 2
  ret void
}
```

The important part that I'm now missing under `compileCreate` is the setup of the vtable (done in `  store ptr getelementptr inbounds ({ [22 x ptr] }, ptr @_ZTVN2DB20SingleValueDataFixedItEE, i32 0, inrange i32 0, i32 2), ptr %3, align 8`) but I have no idea how reproduce something similar via LLVM IR


I had several ideas, but I didn't manage to implement any of them:
* Somehow get a) the pointer to the vtable programatically (currently it's offset 0, but empirical evidence won't help if anything changes in the future) and b) the value of the class virtual pointer (maybe via the name somehow?) and set it up.
* Have an static instantiation of the class `static SingleValueDataFixed<T> example_instantiation;` and do `b.CreateMemCpy` from the static instance to the new one. As `example_instantiation` is already initialized it will copy the object exactly, with the proper virtual table pointer. The problem I found here is that I don't know how to create a LLVM::Value from the static variable to copy from it.

@kitaisreal Do you have of any ideas about how to approach it in a different / better way?

Reply:
Maybe a simpler approach would be to register and call the constructor directly. Something along the lines of:

```c++
{
    llvm::IRBuilder<> & b = static_cast<llvm::IRBuilder<> &>(builder);
    auto * func_declaration_type = llvm::FunctionType::get(b.getVoidTy(), { }, /*isVarArg=*/false);
    auto * func_declaration = llvm::Function::Create(func_declaration_type, llvm::Function::ExternalLinkage, "_ZN2DB20SingleValueDataFixedIjEC2Ev", b.GetInsertBlock()->getModule());
    b.CreateCall(func_declaration, aggregate_data_ptr);
}
```